### PR TITLE
Enable more HTTP tests

### PR DESCRIPTION
**Pull request by eddyashton**
Doing another pass to make sure we're running all currently supported tests for HTTP.

After this PR, there remain 7 HTTP-disabled tests which rely on C++ clients:
```
logging_client_test

member_client_test
voting_history_test
code_update_test

small_bank_client_test
small_bank_sigs_client_test
small_bank_sigs_forwarding
```

Future resolutions for these:
The first uses a custom C++ client which should probably be removed entirely now we're happy with Python clients.
The next 3 use the C++ memberclient - I believe we can change the tests to avoid this once HTTP signatures are complete, perhaps we also replace the C++ memberclient.
The final 3 are perf tests using C++ clients, with HTTP replacements incoming.

### CODE REVIEW

**Comment by jumaffre** [comment]
The `code_update_test` and `member_client_test` tests can be enabled today with minimal tweaking: they only use the `memberclient` C++ utility because I did not previously enable the HTTP equivalent for the `infra.consortium`'s `ack()`, `withdraw()` and `add_new_code()` functions.

The `voting_history_test` really is the only test that makes useful use of the client signatures.

File: CMakeLists.txt [hunk 1]
```diff
@@ -377,39 +379,39 @@ if(BUILD_TESTS)
...
-          --oesignkeypath ${CMAKE_CURRENT_SOURCE_DIR}/src/apps/sample_key.pem
-          # TODO: This test spins up many nodes that go through the join protocol
-          # Since oe_verify_report can take quite a long time to execute (~2s)
-          # and trigger Raft elections, the election timeout should stay high
-          # until https://github.com/microsoft/CCF/issues/480 is fixed
-          --election-timeout 5000
-      )
+      ## Requires C++ memberclient, not currently HTTP compatible
+      if (NOT HTTP)
+        add_e2e_test(
```
Comment (jumaffre) [inline CMakeLists.txt]: As per my comment, we can enable this.

File: cmake/common.cmake [hunk 2]
```diff
@@ -663,6 +663,10 @@ endfunction()
 
 ## Helper for building end-to-end perf tests using the python infrastucture
 function(add_perf_test)
+  ## Perf tests currently use C++ clients which are not sending HTTP - disable
+  if(HTTP)
+    return()
```
Comment (jumaffre) [inline cmake/common.cmake]: Should we write a friendly cmake message for this?

**Comment by cimetrics** [comment]
images

File: cmake/common.cmake (hunk 2 above)
Comment (achamayou) [reply cmake/common.cmake]: Not sure why have this as opposed to if(HTTP) around instances of add_perf_test?

**Comment by eddyashton** [comment]
RE: feedback - have now enabled `member_client_test` (by implementing some extra calls in `consortium.py`, and disabling verification of `ack` signatures), and `code_update_test`. Perf tests are more readably disabled at calling site, rather than in the `add_perf_test` function.

**Comment by jumaffre** [comment]
Is it worth waiting for https://github.com/microsoft/CCF/pull/611 to be merged and then enable all the end-to-end tests?

**Comment by eddyashton** [comment]
> Is it worth waiting for #611 to be merged and then enable all the end-to-end tests?

Yes

**Comment by eddyashton** [comment]
Closing, replaced by #622

**Pull request closed without merging**